app/src/main/java/com/sun/basic_japanese/alphabet/allpages/AlphabetFragment.kt

```diff
@@ -0,0 +1,62 @@
+package com.sun.basic_japanese.alphabet.allpages
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.sun.basic_japanese.R
+import com.sun.basic_japanese.alphabet.eachpage.AlphabetPageFragment
+import com.sun.basic_japanese.alphabet.adapter.PagerAdapter
+import com.sun.basic_japanese.base.BaseFragment
+import com.sun.basic_japanese.constants.BasicJapaneseConstants.HIRAGANA
+import com.sun.basic_japanese.constants.BasicJapaneseConstants.KATAKANA
+import kotlinx.android.synthetic.main.fragment_alphabet.*
+
+class AlphabetFragment : BaseFragment(), AlphabetContract.View {
+    private var alphabetPresenter: AlphabetPresenter? = null
+    private var listener: OnFragmentInteractionListener? = null
+
+    override fun setPresenter(presenter: AlphabetContract.Presenter) {
+        alphabetPresenter = presenter as AlphabetPresenter
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.fragment_alphabet, container, false)
+
+
+    override fun onDetach() {
+        super.onDetach()
+        listener = null
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setPresenter(AlphabetPresenter(this))
+        setupViewPager()
+    }
+
+    private fun setupViewPager() {
+        val gridFragments = ArrayList<AlphabetPageFragment>()
+        gridFragments.apply {
+            add(AlphabetPageFragment.newInstance(HIRAGANA))
+            add(AlphabetPageFragment.newInstance(KATAKANA))
+        }
+        val tabLayoutTitles = ArrayList<CharSequence?>().apply {
```

Comment:
Không nên dùng ArrayList vì nó lưu trữ dạng Array rất phí tài nguyên. Dùng MutableList là đủ để tạo list khả biến rồi.
```suggestion
        val tabLayoutTitles = mutableListOf<CharSequence>().apply {
```